src/modules/kubernetes.rs

```diff
@@ -955,4 +981,135 @@ users: []
         assert_eq!(expected, actual);
         dir.close()
     }
+
+    #[test]
```

Comment:
This set of tests looks good for the change. A number of the tests from #614 seem to be covered by other existing tests at this point.